pkg/operator/ceph/cluster/watcher.go

```diff
@@ -130,6 +145,333 @@ func (c *clientCluster) onK8sNode(ctx context.Context, object runtime.Object) bo
+
+		}
+		// else if len(splitVolumeInUseBased) == 2 && splitVolumeInUseBased[0] == fmt.Sprintf("%s.cephfs.csi.ceph.com", cluster.Namespace) {
+		// 	cephFSVolumeInUse = append(cephFSVolumeInUse, splitVolumeInUseBased[1])
+		// }
+	}
+	return rbdVolumesInUse, cephFSVolumeInUse
+}
+
+func listRBDAndCephFSPV(listPVs *corev1.PersistentVolumeList, cluster *cephv1.CephCluster, rbdVolumesInUse []string) ([]corev1.PersistentVolume, []corev1.PersistentVolume) {
+	var listRbdPV, listCephFSPV []corev1.PersistentVolume
+
+	for _, pv := range listPVs.Items {
+		if pv.Spec.CSI.Driver == fmt.Sprintf("%s.rbd.csi.ceph.com", cluster.Namespace) {
+			for _, rbdVolume := range rbdVolumesInUse {
+				if pv.Spec.CSI.VolumeHandle == rbdVolume {
+					listRbdPV = append(listRbdPV, pv)
+				}
+			}
+		}
+		// else if pv.Spec.CSI.Driver == fmt.Sprintf("%s.cephfs.csi.ceph.com", cluster.Namespace) {
+		// 	for _, cephFSVolume := range cephFSVolumeInUse {
+		// 		if pv.Spec.CSI.VolumeHandle == cephFSVolume {
+		// 			listCephFSPV = append(listCephFSPV, pv)
+		// 		}
+		// 	}
+		// }
+	}
+	return listRbdPV, listCephFSPV
+}
+
+func (c *clientCluster) fenceRbdImage(
+	ctx context.Context, node *corev1.Node, cluster *cephv1.CephCluster,
+	clusterInfo *cephclient.ClusterInfo, rbdPV corev1.PersistentVolume) error {
+
+	logger.Infof("rbd PV NAME %v", rbdPV.Spec.CSI.VolumeAttributes)
+	args := []string{"status", fmt.Sprintf("%s/%s", rbdPV.Spec.CSI.VolumeAttributes["pool"], rbdPV.Spec.CSI.VolumeAttributes["imageName"])}
+	cmd := cephclient.NewRBDCommand(c.context, clusterInfo, args)
+	cmd.JsonOutput = true
+
+	buf, err := cmd.Run()
+	if err != nil {
+		return pkgerror.Wrapf(err, "failed to list watchers for pool/imageName %s/%s.", rbdPV.Spec.CSI.VolumeAttributes["pool"], rbdPV.Spec.CSI.VolumeAttributes["imageName"])
+	}
+
+	logger.Infof("rbd status to get the client ips %v", string(buf))
+	ips, err := rbdStatusUnMarshal(buf)
+	if err != nil {
+		return pkgerror.Wrapf(err, "failed to unmarshal rbd status output")
+	}
+	if len(ips) != 0 {
+		err = c.createNetworkFence(ctx, rbdPV, node, cluster, ips)
+		if err != nil {
+			return pkgerror.Wrapf(err, "failed to create network fence for node %q", node.Name)
+		}
+	}
+
+	return nil
+}
+
+func rbdStatusUnMarshal(output []byte) ([]string, error) {
+	type rbdStatus struct {
+		Watchers []struct {
+			Address string `json:"address"`
+		} `json:"watchers"`
+	}
+
+	var rbdStatusObj rbdStatus
+	err := json.Unmarshal([]byte(output), &rbdStatusObj)
+	if err != nil {
+		return []string{}, pkgerror.Wrapf(err, "failed to unmarshal rbd status output")
+	}
+
+	watcherIPlist := []string{}
+	for _, watcher := range rbdStatusObj.Watchers {
+		// split with separation '/' to remove nounce and concatenating `/32` to define a network with only one IP address
+		address := strings.Split(watcher.Address, "/")[0] + "/32"
+		watcherIPlist = append(watcherIPlist, address)
+	}
+	return watcherIPlist, nil
+}
+
+// func (c *clientCluster) fennceCephFSVolumes(
```

Comment:
are you going to work on it or remove them?

Reply:
I see a lot of commented code

Reply:
blocking is not supported by Networkfence and since I have almost all the code for the cephfs so I  have kept that 😃 . I have mentioned this in the pr description too.

Also, open the issue to track for cephfs too in rook side

Reply:
moved to different branch

Reply:
Great